source/server/connection_handler_impl.cc

```diff
@@ -85,8 +85,12 @@ ConnectionHandlerImpl::SslActiveListener::SslActiveListener(ConnectionHandlerImp
                                  use_proxy_proto, use_orig_dst),
                      factory, socket.localAddress()->asString()) {}
 
-Network::Listener* ConnectionHandlerImpl::findListener(const std::string& socket_name) {
-  auto l = listeners_.find(socket_name);
+Network::Listener* ConnectionHandlerImpl::findListenerByPort(uint32_t port) {
+  auto l = std::find_if(
```

Comment:
nit: given that the number of listeners is going to be tiny (or potentially even 1), I think using a list here is good. Can you just drop a small comment in that we are doing an O(N) scan here that won't work well for a large number of listeners and we may need to use a map in the future.